### PR TITLE
added SingleInstanceUnitTest

### PR DESCRIPTION
**Pull request by cristipp**

### CODE REVIEW

**Comment by cristipp** [comment]
@markschaake

**Comment by markschaake** [comment]
:+1:

File: testkit/src/main/scala/org/allenai/common/testkit/SingeInstanceUnitSpec.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.allenai.common.testkit
+
+/** Base class for any unit test that, for some reason, want to share the Spec instance.
+  * Usually the reason is a slow expensive test setup time (setting up a 3rd party DB, etc.)
+  */
+abstract class SingeInstanceUnitSpec extends AllenAiBaseSpec
```
Comment (cristipp) [inline testkit/src/main/scala/org/allenai/common/testkit/SingeInstanceUnitSpec.scala]: @markschaake @jkinkead Does this start to sound more like an integration test? It's cool we can reuse Scala test infrastructure for both unit and it, I just wonder whether we want to make the distinction more clear via the name of the base classes as well.

Something like IntegrationSpec. <end brainstorm>

**Comment by markschaake** [comment]
Thanks @cristipp ! Merge on successful build?

**Comment by markschaake** [comment]
oops, realized @jkinkead is the assignee

**Comment by cristipp** [comment]
You both guys are awesome :)

File: testkit/src/main/scala/org/allenai/common/testkit/SingeInstanceUnitSpec.scala (hunk 1 above)
Comment (markschaake) [inline testkit/src/main/scala/org/allenai/common/testkit/SingeInstanceUnitSpec.scala]: Would be good to have a `IntegrationSpec` too. I'd rather we kept it separate from the `SingleInstanceUnitSpec`. Could just be :

```scala
abstract class IntegrationSpec extends AllenAiBaseSpec
```

Comment (cristipp) [inline testkit/src/main/scala/org/allenai/common/testkit/SingeInstanceUnitSpec.scala]: Done.

**Comment by jkinkead** [comment]
LGTM! Although you have a failed check of some sort.

**Comment by cristipp** [comment]
Thanks for the feedback.

Is semaphore an alternative for shippable we are looking into? They seem to be broken, the error is "[error] (webapp/compile:compileIncremental) java.lang.NoSuchMethodError: scala.reflect.internal.StdNames$nme$.argument()Lscala/reflect/internal/Names$TermName;"

sbt test works for me.

**Comment by markschaake** [comment]
> Is semaphore an alternative for shippable we are looking into?

@schmmd ?

**Comment by schmmd** [comment]
I'm checking to see how well their caching works out of the box.

**Comment by cristipp** [comment]
Ok, I'll go with Shippable green then :)

**Pull request merged**